pyscraper/sp_2024/parse.py

```diff
@@ -0,0 +1,287 @@
+"""
+This module contains tools to convert the unstructured HTML of the debates into structured XML.
+This is not the TWFY style XML - but tries to retain all information from the original.
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+from bs4 import BeautifulSoup
+from lxml import etree
+
+
+# HTML elements we accept moving from raw_html to parsed
+acceptable_elements = [
+    "a",
+    "abbr",
+    "acronym",
+    "address",
+    "area",
+    "b",
+    "big",
+    "blockquote",
+    "body",
+    "br",
+    "button",
+    "caption",
+    "center",
+    "cite",
+    "code",
+    "col",
+    "colgroup",
+    "dd",
+    "del",
+    "dfn",
+    "dir",
+    "div",
+    "dl",
+    "dt",
+    "em",
+    "font",
+    "form",
+    "head",
+    "h1",
+    "h2",
+    "h3",
+    "h4",
+    "h5",
+    "h6",
+    "hr",
```

Comment:
There are `<hr>`s being included in the output, e.g. start of topical question. And the minor heading is a `<p>` above the first speech, not being spotted - e.g. https://www.parliament.scot/chamber-and-committees/official-report/search-what-was-said-in-parliament/meeting-of-parliament-16-04-2024?meeting=15798&iob=134853 "Free Bikes for School-age Children" and hr is appearing at the end of previous speech by Patrick Harvie, and the one right at the start on new-build heat standard is lost.

Reply:
So we want this preserved as h3 maybe? and then converted into a minor-heading in the final bit?

Reply:
Yeah, sounds good to me - perhaps even a h2/h1 (or better, 'heading'? :) ), there's no headings in the parsed cmpages data after all, the headings are in the committee/agenda_item attributes at present.

Reply:
Added 'heading' support in 5a915f72, converted to minor-heading.